src/external/dftk_to_wannier90.jl

```diff
@@ -0,0 +1,605 @@
+"""
+function overlap_Mmn_k_kpb(pw_basis::PlaneWaveBasis,ψ,k,kpb,n_bands; K_shift = [0,0,0])
+    Mkb = zeros(Float64,n_bands*n_bands,2)
+    current_line = 0 #Manual count necessary to match the format of .mmn files.
+
+    for n in 1:n_bands
+        for m in 1:n_bands
+            
+            ovlp = 0im
+            
+            ## Extract Fourier coeffs and corresponding vectors in reciprocal lattice
+            Gk_coeffs = ψ[k][:,m] 
+            Gk_vec = G_vectors(pw_basis.kpoints[k])
+            Gkpb_coeffs = ψ[kpb][:,n]
+            Gkpb_vec = [ G - K_shift for G in G_vectors(pw_basis.kpoints[kpb]) ] # Don't forget the shift, see the DOC block
+            
+            ## Search for common Fourier modes, corresponding indices are written in map_fourier_mode
+            map_fourier_modes = []
+            for G1 in Gk_vec
+                for G2 in Gkpb_vec
+                    if  G1 == G2
+                        iG1 = only(findall(x-> x==G1,Gk_vec))
+                        iG2 = only(findall(x-> x==G2,Gkpb_vec))
+                        push!(map_fourier_modes,[iG1,iG2])
+                    end
+                end
+            end
+            
+            ## Compute the overlap for mn
+            for (i,j) in map_fourier_modes
+                ovlp += conj(Gk_coeffs[i])*Gkpb_coeffs[j]
+            end
+            
+            current_line += 1 #Go to the next line
+            Mkb[current_line,:] = [real(ovlp),imag(ovlp)]
+
+        end
+    end
+
+    Mkb
+
+ end
+
+
+
+"""
+Iteratively use the preceeding function on each k and kpb to generate the whole .mmn file.
+"""
+function generate_mmn_file(prefix::String,pw_basis::PlaneWaveBasis,ψ, nn_kpts, nn_num)
+    ## general parameters
+    n_bands = size(ψ[1],2)
+    k_size = length(ψ)
+    
+    progress = Progress(size(nn_kpts,1),desc = "Computing Mmn overlaps : ")
+    
+    # Small function for the sake of clarity
+    read_nn_kpts(n) = nn_kpts[n][1],nn_kpts[n][2],nn_kpts[n][3:end]
+
+    ## Write file
+    open("$prefix.mmn","w") do f
+        write(f,"Generated by DFTK at ",string(now()),"\n")
+        write(f,"$n_bands  $k_size  $nn_num"*"\n")
+        
+        for i_nnkp in 1:size(nn_kpts,1) #Loop over all (k_points, neirest_neighbour, shif_vector)
+            #Write the label of the matrix
+            k,kpb,shift = read_nn_kpts(i_nnkp)
+            write(f,string(k)*"  "*string(kpb)*"  "*string(shift[1])*"  "*string(shift[2])*"  "*string(shift[3])*"\n")   
+            #The matrix itself
+            Mkb = overlap_Mmn_k_kpb(pw_basis,ψ,k,kpb,n_bands,; K_shift = shift)
+            for i in 1:n_bands*n_bands
+                write(f, "$(Mkb[i,1])  $(Mkb[i,2])"*"\n")
+            end
+            next!(progress)
+        end
+    end
+
+end
+
+
+"""
+Given a k point, provide the indices of the corresponding G_vectors in the general FFT grid.
+Used in 'A_k_matrix_gaussian_guesses'.
+"""
+function map_k_G_vectors(pw_basis::PlaneWaveBasis,k,fft_grid)
```

Comment:
Don't we have a function for that (index_G_vectors)?

Reply:
I think you're right and it is called..... "index_G_vectors" !